### PR TITLE
Add the download action to the each list item's action menu.

### PR DESCRIPTION
**Pull request by qwangd2l**
Adjust layout to accomodate the dropdown menu.

### CODE REVIEW

File: src/components/content-list/content-list-item.js [hunk 1]
```diff
@@ -8,39 +8,61 @@ import {
...
+	firstUpdated() {
+		super.firstUpdated();
+		this.addEventListener('d2l-dropdown-close', this.dropdownClosed);
+		this.addEventListener('d2l-dropdown-open', this.adjustDropdownBoundary);
+	}
+
 	render() {
 		return html`
-		<div separators="all">
+		<d2l-list separators="all">
```
Comment (ktonon) [inline src/components/content-list/content-list-item.js]: Is this necessary?

File: src/components/content-list/content-list-item.js [hunk 2]
```diff
@@ -87,6 +111,55 @@ class ContentListItem extends DependencyRequester(InternalLocalizeMixin(LitEleme
...
+			e.target.focus();
+			if (opened) {
+				e.target.parentNode.style.zIndex = actionsActiveZIndex;
+			} else {
+				e.target.parentNode.style.zIndex = actionsDefaultZIndex;
+			}
+		}
+	}
+
+	adjustDropdownBoundary() {
```
Comment (qwangd2l) [inline src/components/content-list/content-list-item.js]: I'm doing this to not let the dropdown menu get cut off in the scenario that we have many dropdown menu items.
`topOffsetForDropdownMenu` approximately accounts for the space the dropdown menu takes with the chevron.
`minimumBoundingTop` gives us a way to give the dropdown a minimize size

File: src/components/content-list/content-list-item.js [hunk 3]
```diff
@@ -87,6 +111,55 @@ class ContentListItem extends DependencyRequester(InternalLocalizeMixin(LitEleme
...
+			});
+			const downloadWindow = useNewWindowForDownload(this.type) ?
+				window.open('', '_blank') :
+				window;
+
+			downloadWindow.location.replace(res.value);
+		};
+	}
+
+	dropdownClicked(e) {
```
Comment (AdmiralCoco) [inline src/components/content-list/content-list-item.js]: Is this related to the bug Kevin posted?

Maybe add a comment here linking to the GitHub issue, and to remove this if/when it's fixed?

File: src/components/content-list/content-list-item.js (hunk 1 above)
Comment (qwangd2l) [reply src/components/content-list/content-list-item.js]: The accessibility test for this component (which has `role="listitem"` for `d2l-list-item`) will complain that there is no `role="list"` as a parent. So adding this here will solve that.

File: src/components/content-list/content-list-item.js [hunk 4]
```diff
@@ -87,6 +111,55 @@ class ContentListItem extends DependencyRequester(InternalLocalizeMixin(LitEleme
...
+			const distanceToTop = target.getBoundingClientRect().top;
+			const finalBoundingTop = distanceToTop - rootStore.appTop - topOffsetForDropdownMenu;
+
+			this.dropdownBoundary = {
+				above: Math.max(finalBoundingTop, minimumBoundingTop)
+			};
+		}
+	}
+
+	dropdownClosed() {
```
Comment (AdmiralCoco) [inline src/components/content-list/content-list-item.js]: Ditto - also, would group these two functions together

File: src/components/content-list/content-list-item.js (hunk 3 above)
Comment (qwangd2l) [reply src/components/content-list/content-list-item.js]: Yes that is related to that GitHub issue.
I'll add the comment.
